pkg/operator/controller/nodeobservabilityrun/controller.go

```diff
@@ -87,23 +87,34 @@ func (r *NodeObservabilityRunReconciler) Reconcile(ctx context.Context, req ctrl
 		return
 	}
 
+	defer func() {
+		errUpdate := r.Status().Update(ctx, instance, &client.UpdateOptions{})
+		if errUpdate != nil {
+			r.Log.Error(err, "failed to update status")
+			err = utilerrors.NewAggregate([]error{err, errUpdate})
+		}
+	}()
+
 	// interact with MCO only if we are executing CrioKubeletProfile runs
 	// also bypass if the runtype is e2e-test
 	if instance.Spec.RunType != nodeobservabilityv1alpha1.E2ETest && (instance.Spec.RunType == nodeobservabilityv1alpha1.CrioKubeletProfile) {
-		exists, mco, err := r.ensureMCO(ctx)
-		if err != nil {
+		if err := r.ensureNOMC(ctx); err != nil {
 			return ctrl.Result{}, err
-		} else if !exists {
-			return ctrl.Result{}, fmt.Errorf("could not get NodeObservabilitymachineConfig")
 		}
-		// this will ensure we dont trigger the agent call until the mco and mcp have been updated
-		// successfully
-		result, err := r.MCOReconciler.CheckNodeObservabilityMCPStatus(ctx)
+		// this will ensure we dont trigger the agent call
+		// until the mco and mcp have been updated successfully
+
+		enabled, err := r.checkNOMCStatus(ctx, nodeobservabilityv1alpha1.DebugEnabled)
 		if err != nil {
-			r.Log.Info("waiting for NodeObservabilityMachineConfig update in NodeobeservabilityRun")
-			return result, err
+			r.Log.Error(err, "NodeObservabilityMachineConfig enable status check failed")
+			return ctrl.Result{RequeueAfter: 1 * time.Minute}, err
```

Comment:
Why do you want to requeue in 1 minute?

Reply:
If node labeling, MachineConfig or Pool creation fails then the same is update NOMC status, but if it was successful and MCO is updating the configs it will take more than a minute for it to finish. So thought to reconcile back again after a minute was apt.